docs/faq.rst

```diff
@@ -0,0 +1,225 @@
+
+How does HIP compare with OpenCL?
+---------------------------------
+
+HIP offers several benefits over OpenCL:
+
+* Device code can be written in modern C++, including templates, lambdas,
+  classes and so on.
+* Host and device code can be mixed in the source files.
+* The HIP API is less verbose than OpenCL and is familiar to CUDA developers.
+* Porting from CUDA to HIP is significantly easier than porting from CUDA to OpenCL.
+* HIP uses development tools specialized for each platform: :ref:`amdclang++ <llvm-project:index>`
+  for AMD GPUs or :xref:`nvcc <https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html>`
+  for NVIDIA GPUs, and profilers like :ref:`omniperf <omniperf:index>` or
+  :xref:`Nsight Systems <https://developer.nvidia.com/nsight-systems>`.
+* HIP provides
+  * pointers and host-side pointer arithmetic.
+  * device-level control over memory allocation and placement.
+  * an offline compilation model.
+
+How does porting CUDA to HIP compare to porting CUDA to OpenCL?
+---------------------------------------------------------------
+
+OpenCL differs from HIP and CUDA when considering the host runtime,
+but even more so when considering the kernel code.
+The HIP device code is a C++ dialect, while OpenCL is C99-based.
+OpenCL does not support single-source compilation.
+
+As a result, the OpenCL syntax differs significantly from HIP, and porting tools
+must perform complex transformations, especially when it comes to templates
+or other C++ features in kernels.
+
+To better understand the syntax differences, see :doc:`here<reference/terms>` or
+the :doc:`HIP porting guide <how-to/hip_porting_guide>`.
+
+Can I install CUDA and ROCm on the same machine?
+------------------------------------------------
+
+Yes, but you still need a compatible GPU to actually run the compiled code.
+
+HIP detected my platform incorrectly. What should I do?
+-------------------------------------------------------
+
+See the :doc:`HIP porting guide<how-to/hip_porting_guide>` under the section "Identifying HIP Runtime".
+
+On NVIDIA platforms, can I mix HIP code with CUDA code?
+-------------------------------------------------------
+
+Yes. Most HIP types and data structures are `typedef`s to CUDA equivalents and
+can be used interchangeably.
+
+See the :doc:`how-to/hip_porting_guide` for more details.
+
+Compiler related questions
+==========================
+
+How to use HIP-Clang to build HIP programs?
+------------------------------------------------------
+
+:ref:`hipcc <HIPCC:index>` is a compiler driver. This means it is not a compiler,
+but calls the appropriate compilers and sets some options.
+
+The underlying compilers are :ref:`amdclang++ <llvm-project:index>` or
+:xref:`nvcc <https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html>`,
+depending on the platform, and can be called directly.
+
+What is HIP-Clang?
+------------------
+
+HIP-Clang is a Clang/LLVM-based compiler used to compile HIP programs for AMD
```

Comment:
Is it AMD platforms or ROCM platforms?

Reply:
This one specifically is for AMD platforms. Clang itself also has the ability to compile CUDA code, but that ability is not directly related to "HIP-Clang"